dsc/src/resource_command.rs

```diff
@@ -71,20 +86,29 @@ pub fn get_all(dsc: &mut DscManager, resource: &str, _input: &Option<String>, _s
     }
 }
 
-pub fn set(dsc: &mut DscManager, resource: &str, input: &Option<String>, stdin: &Option<String>, format: &Option<OutputFormat>) {
+/// Set operation.
+///
+/// # Panics
+///
+/// Will panic if provider-based resource is not found.
+///
+pub fn set(dsc: &DscManager, resource_str: &str, input: &Option<String>, stdin: &Option<String>, format: &Option<OutputFormat>) {
     let mut input = get_input(input, stdin);
     if input.is_empty() {
         error!("Error: Input is empty");
         exit(EXIT_INVALID_ARGS);
     }
 
-    let mut resource = get_resource(dsc, resource);
+    let Some(mut resource) = get_resource(dsc, resource_str) else {
+        error!("{}", DscError::ResourceNotFound(resource_str.to_string()).to_string());
+        return
+    };
 
     debug!("resource.type_name - {} implemented_as - {:?}", resource.type_name, resource.implemented_as);
 
-    if let Some(requires) = resource.requires {
-        input = add_type_name_to_json(input, resource.type_name);
-        resource = get_resource(dsc, &requires);
+    if let Some(requires) = &resource.requires {
+        input = add_type_name_to_json(input, resource.type_name.clone());
+        resource = get_resource(dsc, requires).unwrap();
```

Comment:
same here, there generally should not be any `unwrap()` calls outside of unittests

Reply:
Updated